websockets/cext.c

```diff
@@ -0,0 +1,128 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+#include <stdlib.h> /* getenv */
+
+#ifndef HAS_SSE2
+/* setup HAS_SSE2 macro according to compiler predefined macros */
+#ifdef __SSE2__
+#define HAS_SSE2 1
+#else
+#define HAS_SSE2 0
+#endif
+#endif /* not defined HAS_SSE2 */
+
+#if HAS_SSE2
+#include <emmintrin.h>
+#endif
+
+#ifndef HAS_FAST_UNALIGNED_ACCESS
+/* This is not enabled by default */
+/* build with CFLAGS="-DHAS_FAST_UNALIGNED_ACCESS=1" to enable */
+/* There seem to be no information regarding alignment of PyBytes allocated buffer */
+#define HAS_FAST_UNALIGNED_ACCESS 0
+#endif /* not defined HAS_FAST_UNALIGNED_ACCESS */
+
+#if HAS_FAST_UNALIGNED_ACCESS
+#include <stdint.h>
+#endif
+
+static PyObject* apply_mask(PyObject* self, PyObject* args) {
+    const char* mask;
+    const char* in;
+    char* out;
+    Py_ssize_t mask_len;
+    Py_ssize_t len;
+    Py_ssize_t i = 0; /* initialize iterator */
+    PyObject* result;
+
+
+    if (!PyArg_ParseTuple(args, "s#s#", &mask, &mask_len, &in, &len)) {
+        return NULL;
+    }
+
+    result = PyBytes_FromStringAndSize(NULL, len);
```

Comment:
You will read outside the boundaries of the input buffer. If you cross a page boundary this can crash. So once again, back to the comment "there seem to be no normative information regarding alignment of PyBytes allocated buffer."